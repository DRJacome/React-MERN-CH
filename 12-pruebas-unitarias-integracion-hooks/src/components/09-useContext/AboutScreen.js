import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const AboutScreen = () => {
  const { usuario, setUsuario } = useContext(Contexto)
  const gestionarClick = () => {
    setUsuario({})
  }
  return (
    <div>
      <h1>Acerca de</h1>
      <hr />
      <pre>{JSON.stringify(usuario, null, 3)}</pre>
      <button className='btn btn-primary' onClick={gestionarClick}>Cerrar sesi√≥n</button>
    </div>
  )
}