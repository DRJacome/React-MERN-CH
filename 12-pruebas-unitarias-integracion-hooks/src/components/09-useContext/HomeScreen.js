import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const HomeScreen = () => {
  const { usuario } = useContext(Contexto)
  console.log(usuario);
  return (
    <div>
      <h1>Pantalla de inicio</h1>
      <hr />
      <pre>{JSON.stringify(usuario, null, 3)}</pre>
    </div>
  )
}