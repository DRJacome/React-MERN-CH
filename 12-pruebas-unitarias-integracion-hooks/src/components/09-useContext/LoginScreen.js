import React, { useContext } from 'react'
import { Contexto } from './UserContext'

export const LoginScreen = () => {
  // 1. Obtener la referencia al userContext.
  const { setUsuario } = useContext(Contexto)
  // 2. setUser.
  /*  {
     id: 123,
     nombre: 'David'
   } */

  return (
    <div>
      <h1>Pantalla de Login</h1>
      <hr />
      <button className='btn btn-primary' onClick={() => {
        setUsuario({
          id: 123,
          nombre: 'David'
        })
      }}>Login</button>
    </div>
  )
}