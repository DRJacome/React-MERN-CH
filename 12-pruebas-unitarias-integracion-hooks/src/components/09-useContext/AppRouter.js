import React from "react";
/* Para React Router v5.x -> usar Switch;
* Para React Router V6.x -> usar Routes.
 */
import {
  BrowserRouter as Router,
  Switch, //Routes,
  Route
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";
import { Redirect } from "react-router-dom";


export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Redirect to='/' /> {/* <Route component={HomeScreen} /> */}
          </Switch>
        </div>
      </div>
    </Router>
  )
}