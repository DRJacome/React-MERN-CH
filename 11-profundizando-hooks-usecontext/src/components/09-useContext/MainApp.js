import { useState } from "react"
import { AppRouter } from "./AppRouter"
import { Contexto } from "./UserContext"

export const MainApp = () => {
  const [usuario, setUsuario] = useState({})
  return (
    <Contexto.Provider value={{
      usuario,
      setUsuario
    }}>
      <AppRouter />
    </Contexto.Provider>
  )
}