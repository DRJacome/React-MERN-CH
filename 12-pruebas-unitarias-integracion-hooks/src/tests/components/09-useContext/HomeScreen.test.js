import React from 'react'
import { mount } from 'enzyme'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { Contexto } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen/>', () => {

  const usuario = {
    nombre: 'David',
    email: 'david@gmail.com'
  }
  const wrapper = mount(
    <Contexto.Provider value={{
      usuario
    }}>
      <HomeScreen />
    </Contexto.Provider>
  );

  test('Debe renderizarse correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  })
})