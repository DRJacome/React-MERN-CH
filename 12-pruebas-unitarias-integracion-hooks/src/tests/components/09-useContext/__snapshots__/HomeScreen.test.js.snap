// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> Debe renderizarse correctamente 1`] = `
<div>
  <h1>
    Pantalla de inicio
  </h1>
  <hr />
  <pre>
    {
   "nombre": "David",
   "email": "david@gmail.com"
}
  </pre>
</div>
`;
